op-node/benchmarks/batchbuilding_test.go

```diff
@@ -0,0 +1,233 @@
+package benchmarks
+
+import (
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-batcher/compressor"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+
+	// compressors used in the benchmark
+	rc, _ = compressor.NewRatioCompressor(compressor.Config{
+		TargetOutputSize: 100_000_000_000,
+		ApproxComprRatio: 0.4,
+	})
+	sc, _ = compressor.NewShadowCompressor(compressor.Config{
+		TargetOutputSize: 100_000_000_000,
+	})
+	nc, _ = compressor.NewNonCompressor(compressor.Config{
+		TargetOutputSize: 100_000_000_000,
+	})
+
+	compressors = map[string]derive.Compressor{
+		"NonCompressor":    nc,
+		"RatioCompressor":  rc,
+		"ShadowCompressor": sc,
+	}
+
+	// batch types used in the benchmark
+	batchTypes = []uint{
+		derive.SpanBatchType,
+		// uncomment to include singular batches in the benchmark
+		//derive.SingularBatchType,
```

Comment:
Do we intend to leave this commented out?

Reply:
Yes, seeing as there isn't much point to testing Singular Batches besides to compare with Span Batches as desired.